### PR TITLE
pkg/steps/artifacts: Wait for running artifacts container

### PR DESCRIPTION
**Pull request by alvaroaleman**
Supersedes and thereby closes https://github.com/openshift/ci-tools/pull/958

### CODE REVIEW

File: pkg/steps/multi_stage_test.go [hunk 1]
```diff
@@ -547,6 +548,9 @@ func TestRun(t *testing.T) {
 }
 
 func TestArtifacts(t *testing.T) {
+	timeSecond = time.Nanosecond
+	defer func() { timeSecond = time.Second }()
```
Comment (bbguimaraes) [inline pkg/steps/multi_stage_test.go:552]: Shouldn't this result in an error?

Comment (alvaroaleman) [reply pkg/steps/multi_stage_test.go:552]: It does but we only log it: https://github.com/openshift/ci-tools/pull/1030/files#diff-892295443de5592354f856e247809216R452

Comment (alvaroaleman) [reply pkg/steps/multi_stage_test.go:552]: github is so great, doesn't expand the are the link is in, this one should work: https://github.com/openshift/ci-tools/blob/77d4e0ac850628341ffc1075527bee60a4d9dbc5/pkg/steps/artifacts.go#L452

File: pkg/steps/artifacts.go [hunk 2]
```diff
@@ -216,6 +217,36 @@ type PodClient interface {
...
+var timeSecond = time.Second
+
+func waitForContainer(podClient PodClient, ns, name, containerName string) error {
+	logrus.WithFields(logrus.Fields{
+		"namespace": ns,
+		"name":      name,
+		"container": containerName,
+	}).Trace("Waiting for container to be running.")
+
+	return wait.PollImmediate(time.Second, 30*timeSecond, func() (bool, error) {
```
Comment (stevekuznetsov) [inline pkg/steps/artifacts.go:230]: Could e.g. volume attach take longer than 30s? Do we care?

Comment (alvaroaleman) [reply pkg/steps/artifacts.go:230]: Probably, the 30 seconds was a random guess. Failures here are non-terminal though, we just log them. 30 seconds is definitely better than not waiting at all. We can use ci-search to figure out if this is too little later on and if yes finetune

**Pull request merged**